### ISSUE TITLE
import time for pgmpy

### ISSUE THREAD

**Issue by palashahuja** [pgmpy/pgmpy#497]
Importing pgmpy takes a lot of time compared to other libraries. Here are the details.

```
time python -c "import pgmpy"
real    0m10.113s
user    0m1.088s
sys 0m0.544s
```

For example:- for numpy the import time is as follows

```
time python -c "import numpy"
real    0m0.284s
user    0m0.216s
sys 0m0.360s
```

(The primary difference between import times is because numpy's core modules are written in C, that's why the efficiency is clearly visible. But we could still look for ways to reduce this time.)

**Comment by khalibartan** [comment]
Agreed. Time taken by pgmpy to import is three times more than libraries which are written in python like sympy or sklearn.

**Comment by ankurankan** [comment]
Ran cProfile on import and looks like `BayesianModel.py` is the culprit, but have no idea why.

**Comment by khalibartan** [comment]
Yeah from pgmpy the most slow call was of `BayesianModel.py` but I believe that total overall time is increased by dependencies of pgmpy, because it has around 4-5 dependencies.
